Break out @luma.gl/webgl module

ibgreen opened the pull request:
<!-- For feature, feature enhancement or bug fix, create an issue first and finish To Do List there -->
<!-- Anything doesn't work as expected is a bug, including code, doc and test -->
# For #724
<!-- For other PRs without open issue -->
#### Background

#### Change List
- @luma.gl/webgl2: This breaks out our webgl classes into a separate module.
    - Splits out half the code from the huge `core` module
    - Clarifies how the luma.gl code base is organized.
- `@luma.gl/core`: reexport all `webgl` functions
    - so apps won't notice the split
    - just like previous shadertools and state-tracker splits
- `@luma.gl/webgl2-state-tracker`:
    - just renames this module to `webgl2-...`
    - ensure all our `webgl2...` modules are consistently names
- Notes:
    - This particular split does not give us any immediate bundle size benefits
    - since `core` will still depend on this module.
- Challenges:
    -  some utility functions needed to be duplicated or shared/reexported.

tsherif commented:
Looks fine, but I'm just curious about the "WebGL 2" naming. Feel like it gives the impression that these are WebGL2-only.

ibgreen commented:
> Feel like it gives the impression that these are WebGL2-only.

Well, the impression I wanted to emphasize is that these libraries fully support WebGL2. I personally think that is what makes them stand out. 

That said, `@luma.gl/webgl2` will run on WebGL1, but only if `webgl2-polyfill` is installed, so in that sense I suppose the impression you have is partially true.

ibgreen commented:
Now rebased on top of the new import linter rules, which gives additional peace of mind when doing this type of splits...

>  just curious about the "WebGL 2" naming. Feel like it gives the impression that these are WebGL2-only.

I reverted this change for now. It does seems like something we should decide when we have a quorum.
